### ISSUE TITLE
bugs and features

### ISSUE THREAD

**Issue by yonderehco1197** [ZL-Audio/ZLEqualizer#64]
Hi!

I would like to start by saying thanks for the regular updates to this great product. In addition to adding useful customizations, the performance and responsiveness have really improved and it is now quite comfortable to use. Thank you!

I also found a few bugs and wanted to suggest a few improvements:

1. When clicking on the analyzer values pre, post, side (quick change or using the dropdown menu) the plugin/DAW freezes/crashes.

2. Sometimes the equalizer curve just stops displaying.

3. After several color scheme changes, the ability to adjust values with the mouse wheel is lost. After reopening the plugin everything returns to normal.

4. Returning to the first bug - is it possible to add resetting of the analyzer graph, for example, on a single click on it?

5. Is it possible to add a permanent display of parameter values? Perhaps as a separate option.

6. And lastly, what about zero delay? :) Right now it is 1 ms even in minimum phase mode.

Windows 10/Studio One 6.6.0/VST3 0.3.0

**Comment by zsliu98** [comment]
> When clicking on the analyzer values pre, post, side (quick change or using the dropdown menu) the plugin/DAW freezes/crashes.
Sometimes the equalizer curve just stops displaying.

I will look into these issues. However, these might be very complex.

> After several color scheme changes, the ability to adjust values with the mouse wheel is lost. After reopening the plugin everything returns to normal.

Same as #63. A fix has been pushed and will be included in `0.3.1`.

> Returning to the first bug - is it possible to add resetting of the analyzer graph, for example, on a single click on it?

Could you please describe the reason for resetting the analyzer?

> Is it possible to add a permanent display of parameter values? Perhaps as a separate option.

Could you please describe the reason for the permanent display? Does the exact freq/gain/Q value help you?

> And lastly, what about zero delay? :) Right now it is 1 ms even in minimum phase mode.

That feature might be included in `0.4.0`. Currently, the dynamic feature processes the audio in 1ms blocks. I may let you adjust the length of the block. Therefore, when the length is zero(per-sample), the latency is zero.

**Comment by yonderehco1197** [comment]
> Could you please describe the reason for resetting the analyzer?

I think I forgot to specify that this applies only to the analyzer in the FRZ position. Sometimes you just need to make the sound visually flat, but the analyzer graph accumulates only once, and a quick reset would make it much easier.

> Could you please describe the reason for the permanent display? Does the exact freq/gain/Q value help you?

It will help to navigate the graph much more confidently and quickly. When I see a point on the graph and all I can say about it is “well, it's somewhere between 2k and 5k with 4 to 8 dB gain” - it's not very convenient :)

**Comment by zsliu98** [comment]
> I think I forgot to specify that this applies only to the analyzer in the FRZ position. Sometimes you just need to make the sound visually flat, but the analyzer graph accumulates only once, and a quick reset would make it much easier.

I need to think about it. Some people may not want that resetting when they click the spectrum.

> It will help to navigate the graph much more confidently and quickly. When I see a point on the graph and all I can say about it is “well, it's somewhere between 2k and 5k with 4 to 8 dB gain” - it's not very convenient :)

A commit has just been pushed. I will also add an OFF/ON option to the UI setting panel.

<img width="475" alt="Screenshot 2024-05-30 at 16 25 04" src="https://github.com/ZL-Audio/ZLEqualizer/assets/35594091/b46d789c-62ea-41c5-b93c-5e5b618c151c">

**Comment by yonderehco1197** [comment]
> I need to think about it. Some people may not want that resetting when they click the spectrum.

Maybe another OFF/ON option somewhere in the Analyzer tab? Anyway, thanks for taking the time to listen to arguments like mine :)

> A commit has just been pushed. I will also add an OFF/ON option to the UI setting panel.

It's a very elegant solution. Thank you!

**Comment by zsliu98** [comment]
Hi! The pre-release 0.3.1 is ready. It should fix the mouse-wheel issue and add the parameter value display (to enable that, you have to go to the UI setting panel and set the opacity of the tag colour to non-zero).
However, I am not sure whether the first two issues are fixed (99% chance they are not :cry:

**Comment by yonderehco1197** [comment]
Mouse wheel bug and EQ curve randomly disappearing have been fixed. The display of parameter values works great. Thank you!
I've encountered literally a few extremely minor problems:

- When starting a new (empty) instance, the parameters for a point with values of 1000 kgz, 0.00 dB are displayed.
- When a point is deleted, its parameters are not deleted with it, but continue to be displayed until a new point is created or an empty spot on the graph is clicked. However, after reopening the values are displayed again.
- When deleting one point and creating another, the new point is created as if “not selected” (no white frame) and, therefore, its parameters are not displayed.

The error with DAW crash when quickly changing the analyzer settings remains, and the same error occurs when quickly turning [collision on/off](https://github.com/ZL-Audio/ZLEqualizer/issues/58). Perhaps this similarity will somehow help with the fix :)

**Comment by yonderehco1197** [comment]
Mouse wheel bug and EQ curve randomly disappearing have been fixed. The display of parameter values works great. Thank you!
I've encountered literally a few extremely minor problems:

- When starting a new (empty) instance, the parameters for a point with values of 1000 Hz, 0.00 dB are displayed.
- When a point is deleted, its parameters are not deleted with it, but continue to be displayed until a new point is created or an empty spot on the graph is clicked. However, after reopening the values are displayed again.
- When deleting one point and creating another, the new point is created as if “not selected” (no white frame) and, therefore, its parameters are not displayed.

The error with DAW crash when quickly changing the analyzer settings remains, and the same error occurs when quickly turning [collision on/off](https://github.com/ZL-Audio/ZLEqualizer/issues/58). Perhaps this similarity will somehow help with the fix :)

**Comment by zsliu98** [comment]
> The error with DAW crash when quickly changing the analyzer settings remains, and the same error occurs when quickly turning https://github.com/ZL-Audio/ZLEqualizer/issues/58. Perhaps this similarity will somehow help with the fix :)

Thanks for your help. I will inform you once an update is available.

> When starting a new (empty) instance, the parameters for a point with values of 1000 Hz, 0.00 dB are displayed.
When a point is deleted, its parameters are not deleted with it, but continue to be displayed until a new point is created or an empty spot on the graph is clicked. However, after reopening the values are displayed again.
When deleting one point and creating another, the new point is created as if “not selected” (no white frame) and, therefore, its parameters are not displayed.

The first two issues will also be fixed.

**Comment by zsliu98** [comment]
Hi. An update is available at https://github.com/ZL-Audio/ZLEqualizer/releases/tag/0.3.1 (the pre-release without *Pre Release* in its name) Those *minor* problems have been fixed. However, I am still not very confident about the one that crashes the DAW (analyzer setting & collision setting).

**Comment by yonderehco1197** [comment]
This fixed absolutely all bugs (including analyzer/collision settings). Awesome! Thank you so much!
I hope I'm not overusing wishes within one issue, but I have literally one more left:

When I reopen the plugin, the last used point is automatically selected (even if I reset the selection). Is it possible to add an automatic reset of the selection, regardless of whether a point has been selected before?

Sorry for not reporting this right away :(

**Comment by zsliu98** [comment]
> This fixed absolutely all bugs (including analyzer/collision settings). Awesome! Thank you so much! I hope I'm not overusing wishes within one issue, but I have literally one more left:
> 
> When I reopen the plugin, the last used point is automatically selected (even if I reset the selection). Is it possible to add an automatic reset of the selection, regardless of whether a point has been selected before?
> 
> Sorry for not reporting this right away :(

Thanks for the feedback. However, I would prefer to set the point as selected so that you can know which filter corresponds to the current band.

I am going to close this issue. Feel free to open another issue/discussion if you have any ideas.
